### PR TITLE
RIDEA-1376 Add support for Garmin Connect and Strava pairing

### PR DESCRIPTION
**Pull request by SzasznikaJanos**

### CODE REVIEW

File: src/com/zone5cloud/core/thirdpartyconnections/connections/ConectionApp.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package com.zone5cloud.core.thirdpartyconnections.connections;
+
+import com.google.gson.annotations.SerializedName;
+
+public class ConectionApp {
```
Comment (ghost) [inline src/com/zone5cloud/core/thirdpartyconnections/connections/ConectionApp.java]: typo in `Conection`

File: src/com/zone5cloud/core/thirdpartyconnections/connections/ConectionApp.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+	@SerializedName("company")
+	private ConnectionCompany company;
+
+	@SerializedName("user")
+	private ConnectionUser connectionUser;
```
Comment (ghost) [inline src/com/zone5cloud/core/thirdpartyconnections/connections/ConectionApp.java]: this should be in `ConnectionsResponse`

File: src/com/zone5cloud/core/thirdpartyconnections/connections/ConnectionInitResponse.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+public class ConnectionInitResponse {
+
+    @SerializedName("url")
+    private String confirmationUrl;
+
+    public String getConfirmationUrl() {
+        return confirmationUrl;
+    }
+
+    public void setConfirmationUrl(String confirmationUrl) {
```
Comment (ghost) [inline src/com/zone5cloud/core/thirdpartyconnections/connections/ConnectionInitResponse.java]: is this needed?

File: src/com/zone5cloud/core/thirdpartyconnections/connections/ConnectionUser.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package com.zone5cloud.core.thirdpartyconnections.connections;
+
+import com.google.gson.annotations.SerializedName;
+
+public class ConnectionUser {
```
Comment (ghost) [inline src/com/zone5cloud/core/thirdpartyconnections/connections/ConnectionUser.java:5]: `id` and `isPremium` fields are missing, are those needed?

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
Indentation is off in a couple of classes.
The docs can be found here
Found the doc in https://portal.zone5cloud.com/en/manager/api-documentation/whats-prod/api/api_reference_connections#get-user-connections with a mock response
```
{
        "user": {
            "id": 0000000,
            "email": "user@example.com",
            "firstname": "Firstname",
            "lastname": "Lastname",
            "isPremium": true
        },
        "type": "garminconnect",
        "enabled": false,
        "app": {
            "company": {
                "cls": "company",
                "name": "Garmin",
                "url": "https://www.garmin.com",
                "privacyUrl": "https://www.garmin.com/en-US/privacy/connect/policy/"
            }
        }
    }
```

File: src/com/zone5cloud/core/thirdpartyconnections/connections/ConnectionInitResponse.java (hunk 3 above)
Comment (SzasznikaJanos) [reply src/com/zone5cloud/core/thirdpartyconnections/connections/ConnectionInitResponse.java]: yes, it is needed for Retrofit, the access modifier is private, in this case retrofit requires a open/public modifier access

File: src/com/zone5cloud/core/thirdpartyconnections/connections/ConectionApp.java (hunk 2 above)
Comment (SzasznikaJanos) [reply src/com/zone5cloud/core/thirdpartyconnections/connections/ConectionApp.java]: Retrofit Good practices should avoid nested classes. It is better to use classes on its own.
the nested classes constructors trying to initialize the outer classes fields are a bit sketchy inside Retrofit converter factories.

File: src/com/zone5cloud/core/thirdpartyconnections/connections/ConectionApp.java (hunk 1 above)
Comment (SzasznikaJanos) [reply src/com/zone5cloud/core/thirdpartyconnections/connections/ConectionApp.java]: Corrected it. Thankss

File: src/com/zone5cloud/core/thirdpartyconnections/connections/ConnectionUser.java (hunk 4 above)
Comment (SzasznikaJanos) [reply src/com/zone5cloud/core/thirdpartyconnections/connections/ConnectionUser.java:5]: Indeed, good catch, i have added the missing fields

File: src/com/zone5cloud/core/thirdpartyconnections/ThirdParty.java [hunk 5]
```diff
@@ -6,6 +6,11 @@
 	public static final String REM_THIRD_PARTY_CONNECTION = "/rest/users/connections/api/v1/live_activities/delete_third_party_token";
```
Comment (jeanhall) [inline src/com/zone5cloud/core/thirdpartyconnections/ThirdParty.java:6]: remove the 3 old endpoints that you are replacing.

**Review by jeanhall — CHANGES_REQUESTED** [review]

**Review by jeanhall — APPROVED** [review]

**Pull request merged**